### PR TITLE
Document `os.rename` and `os.getenv` APIs

### PR DESCRIPTION
**Pull request by tritao**
**What does this PR do?**

Documents the `os.rename` and `os.getenv` APIs

**Did you check all the boxes?**

- [x] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [x] Add unit tests showing fix or feature works; all tests pass
- [x] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [x] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] Minimize the number of commits
- [x] Align [documentation](https://github.com/premake/premake-core/tree/master/website) to your changes

### CODE REVIEW

**Review by Jarod42 — APPROVED** [review]

File: website/docs/os.getenv.md
Comment (samsinsane) [inline website/docs/os.getenv.md:1]: This function comes from the Lua runtime, I don't think we normally document those functions unless we modify them. On one hand, I can see the value in including functions like `os.getenv`, but on the other, I don't know where we draw the line on documenting Lua functions.

File: website/docs/os.rename.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+Rename file system files or directories.
+
+```lua
+os.rename("path", "newpath")
+```
+
+### Parameters ###
+
+`path` is the file system path to be renamed.
```
Comment (samsinsane) [inline website/docs/os.rename.md:9]: Missing `newpath`.

File: website/docs/os.rename.md
Comment (samsinsane) [inline website/docs/os.rename.md:1]: This is similar to the above, though we have extended `os.rename` to support paths with wide characters on Windows. Is that enough to warrant documenting the API?

File: website/sidebars.js [hunk 2]
```diff
@@ -399,6 +400,7 @@ module.exports = {
 						'os.outputof',
 						'os.pathsearch',
 						'os.realpath',
+						'os.rename',
 						'os.remove',
```
Comment (samsinsane) [inline website/sidebars.js:404]: These are around the wrong way.

**Review by Jarod42 — APPROVED** [review]

**Review by samsinsane — APPROVED** [review]
I'm going to approve and merge this, though I'm not sure I agree with documenting Lua APIs like this, but I do agree we should have _something_ for them.

**Pull request merged**